utils/orchestrators.ts

```diff
@@ -132,9 +133,21 @@ export const terminateUpdateCgnOrchestratorTask = (
   fiscalCode: FiscalCode,
   status: string,
   reason: NonEmptyString
-) =>
-  tryCatch(
-    () =>
-      client.terminate(makeUpdateCgnOrchestratorId(fiscalCode, status), reason),
-    toError
+) => {
+  const orchestratorId = makeUpdateCgnOrchestratorId(fiscalCode, status);
+  const voidTask = taskEither.of<Error, void>(void 0);
+  return tryCatch(() => client.getStatus(orchestratorId), toError).chain(
+    maybeStatus =>
+      fromNullable(maybeStatus).foldL(
+        () => voidTask,
+        () =>
+          tryCatch(
+            () => client.terminate(orchestratorId, reason),
+            toError
+          ).foldTaskEither(
+            () => voidTask,
+            () => voidTask
+          )
+      )
```

Comment:
Is all this to avoid terminate an orchestrator which is not running? What does happen if you try to terminate an orchestrator which is not running?

Reply:
Yep, when trying to terminate an orchestrator which is not running, the client raises an error so we can handle it as a no effect, or even the termination has just be done :)

Reply:
Are we checking here a running orchestrator or an orchestrator that has a status?

Reply:
A running orchestrator, btw if it is running it should have also a status

Reply:
If we want terminate only running orchestator we should use this:
```typescript
fromNullable(maybeStatus)
  .chain(
    fromPredicate(
      _ =>
        _.runtimeStatus === df.OrchestrationRuntimeStatus.Running ||
        _.runtimeStatus === df.OrchestrationRuntimeStatus.Pending
    )
)
```